src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -52,13 +52,15 @@ def rescale_stride(stride, ratio):
     return new_strides
 
 
-def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right):
+def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right, dtype=None):
     inputs_len = inputs.shape[0]
     step = chunk_len - stride_left - stride_right
     for i in range(0, inputs_len, step):
         # add start and end paddings to the chunk
         chunk = inputs[i : i + chunk_len]
         processed = feature_extractor(chunk, sampling_rate=feature_extractor.sampling_rate, return_tensors="pt")
+        if dtype is not None:
+            processed = {k: v.to(dtype=dtype) for k, v in processed.items()}
```

Comment:
I believe you can call the `to` directly on `processed`, which is a `BatchFeature` and handles `dtype` in its `to` method thanks to #20536 (was designed for vision but I think it will apply here too).

Reply:
Done !